spring-cloud-skipper-server/src/main/java/org/springframework/cloud/skipper/deployer/AppDeployerReleaseManager.java

```diff
@@ -107,34 +112,67 @@ public Release install(Release releaseInput) {
 		return status(this.releaseRepository.save(release));
 	}
 
+
 	public Release status(Release release) {
 		AppDeployer appDeployer = this.deployerRepository.findByNameRequired(release.getPlatformName())
 				.getAppDeployer();
 		Set<String> deploymentIds = new HashSet<>();
 		AppDeployerData appDeployerData = this.appDeployerDataRepository
 				.findByReleaseNameAndReleaseVersion(release.getName(), release.getVersion());
 		deploymentIds.addAll(StringUtils.commaDelimitedListToSet(appDeployerData.getDeploymentData()));
+		log.debug("Getting status for {} using deploymentIds {}", release,
+				StringUtils.collectionToCommaDelimitedString(deploymentIds));
+
 		if (!deploymentIds.isEmpty()) {
-			boolean allDeployed = true;
-			StringBuffer releaseStatusMsg = new StringBuffer();
+			// mainly track deployed and unknown statuses. for any other
+			// combination, get more details from instances.
+			int deployedCount = 0;
+			int unknownCount = 0;
+			List<String> platformStatusMessages = new ArrayList<>();
 			for (String deploymentId : deploymentIds) {
 				AppStatus appStatus = appDeployer.status(deploymentId);
+				log.debug("App Deployer for deploymentId {} gives status {}", deploymentId, appStatus);
+
+				switch (appStatus.getState()) {
+				case deployed:
+					deployedCount++;
+					break;
+				case unknown:
+					unknownCount++;
+					break;
+				case deploying:
+				case undeployed:
+				case partial:
+				case failed:
+				case error:
+					StringBuffer statusMsg = new StringBuffer(deploymentId + "=[");
+					for (AppInstanceStatus appInstanceStatus : appStatus.getInstances().values()) {
+						statusMsg.append(appInstanceStatus.getId() + "=" + appInstanceStatus.getState());
+					}
+					statusMsg.append("]");
+					platformStatusMessages.add(statusMsg.toString());
+					break;
+				default:
+					break;
+				}
+
 				if (appStatus.getState() != DeploymentState.deployed) {
 					StringBuffer statusMsg = new StringBuffer(deploymentId + "=[");
-					allDeployed = false;
 					for (AppInstanceStatus appInstanceStatus : appStatus.getInstances().values()) {
 						statusMsg.append(appInstanceStatus.getId() + "=" + appInstanceStatus.getState());
 					}
 					statusMsg.append("]");
-					releaseStatusMsg.append(statusMsg);
 				}
 			}
-			if (allDeployed) {
+			if (deployedCount == deploymentIds.size()) {
 				release.getInfo().getStatus().setPlatformStatus("All the applications are deployed successfully.");
 			}
+			else if (unknownCount == deploymentIds.size()) {
+				release.getInfo().getStatus().setPlatformStatus("All applications unknown to system.");
+			}
 			else {
-				release.getInfo().getStatus().setPlatformStatus(
-						"Applications deploying... " + releaseStatusMsg.toString());
+				release.getInfo().getStatus().setPlatformStatus("State of applications: "
```

Comment:
We should give some indication that there maybe apps that are in the state deployed and unknown.  In the work for 'smart update/rollback', I've changed what is stored `appDeployerData.getDeploymentData()` to be a Map of appName, deploymentId which makes it clear what application name is in what state.  In the meantime, maybe we can gather up `appInstanceStatus.getId()` for all the deployed apps and show it in the case of 'some apps deployed, some apps still deploying`.